### PR TITLE
Documentation updates from Docbook to CommonMark

### LINKED ISSUE #104968: Convert Ocaml from Docbook to Commonmark
This issue tracks the conversion of `/doc/languages-frameworks/ocaml.xml` from Docbook to Commonmark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Instructions for performing this conversion

1. Post in this issue that you intend to work on it. (Please also post if you stop working on it)
2. Checkout a new branch of nixpkgs master.
3. Edit `doc/languages-frameworks/index.xml`: change `ocaml.xml` to `ocaml.section.xml`.
4. Move `doc/languages-frameworks/ocaml.xml` to `doc/languages-frameworks/ocaml.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Follow [conversion best practices](https://github.com/NixOS/nixpkgs/issues/105243). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### LINKED ISSUE #105261: Convert Shell-Helpers from Docbook to CommonMark
This issue tracks the conversion of `/doc/builders/packages/shell-helpers.xml` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Conversion instructions

1. Comment here that you intend to work on it. (Please also post if you stop working on it.)
2. Checkout a new branch of nixpkgs master.
3. Edit `doc/builders/packages/index.xml`: change `shell-helpers.xml` to `shell-helpers.section.xml`.
4. Rename `doc/builders/packages/shell-helpers.xml` to `shell-helpers.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Follow [conversion best practices](https://github.com/NixOS/nixpkgs/issues/105243). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

OCaml
https://github.com/NixOS/nixpkgs/issues/104968

Shell-helpers
https://github.com/NixOS/nixpkgs/issues/105261

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Yes, I am pulling various at once.

File: doc/builders/packages/shell-helpers.section.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Interactive shell helpers {#sec-interactive-shell-helpers}
```
Comment (ryantm) [inline doc/builders/packages/shell-helpers.section.md:1]:
```suggestion
# Interactive shell helpers {#sec-shell-helpers}
```
please preserve old anchors.

File: doc/languages-frameworks/ocaml.section.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Ocaml {sec-ocaml}
```
Comment (ryantm) [inline doc/languages-frameworks/ocaml.section.md:1]:
```suggestion
# Ocaml {#sec-language-ocaml}
```

**Review by ryantm — CHANGES_REQUESTED** [review]

File: doc/languages-frameworks/ocaml.section.md (hunk 2 above)
Comment (AndersonTorres) [reply doc/languages-frameworks/ocaml.section.md]: Done

File: doc/builders/packages/shell-helpers.section.md (hunk 1 above)
Comment (AndersonTorres) [reply doc/builders/packages/shell-helpers.section.md]: Done

File: doc/languages-frameworks/ocaml.section.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Ocaml {#sec-language-ocaml}
```
Comment (ryantm) [inline doc/languages-frameworks/ocaml.section.md:1]:
```suggestion
# OCaml {#sec-language-ocaml}
```

Comment (ryantm) [reply doc/languages-frameworks/ocaml.section.md:1]: Sorry, I missed this earlier.

**Review by ryantm — CHANGES_REQUESTED** [review]

File: doc/languages-frameworks/ocaml.section.md (hunk 3 above)
Comment (AndersonTorres) [reply doc/languages-frameworks/ocaml.section.md:1]: Ops!

Comment (AndersonTorres) [reply doc/languages-frameworks/ocaml.section.md]: @ryantm done

**Review by ryantm — APPROVED** [review]
Code and rendering looks good!

**Comment by ryantm** [comment]
Thanks!

**Pull request merged**